### PR TITLE
Add sprite that can be controlled by player

### PR DESCRIPTION
**Pull request by NathanielLow**
Redo of #4 because I'm too lazy to resolve merge conflicts.

### CODE REVIEW

File: Assets/Tests/TestBulletPatterns.cs [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
 using UnityEngine;
 using UnityEngine.TestTools;
 
 namespace Tests
 {
+    [TestFixture]
     public class TestBulletPatterns
     {
-        public Vector3[] getBulletPositions(Bullet[] bullets)
+        private static Vector3[] getBulletPositions(IReadOnlyList<Bullet> bullets)
```
Comment (ProgrammingIncluded) [inline Assets/Tests/TestBulletPatterns.cs:13]: Can you link the documentation for where you found this fixture? Thanks.

File: Assets/Tests/TestBulletPatterns.cs [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
         {
-            Vector3[] positions = new Vector3[bullets.Length];
-            for(int x = 0; x < bullets.Length; ++x)
+            Vector3[] positions = new Vector3[bullets.Count];
+            for(int x = 0; x < bullets.Count; ++x)
```
Comment (ProgrammingIncluded) [inline Assets/Tests/TestBulletPatterns.cs:16]: Isn't Count() a function? http://programminghave.blogspot.com/2014/02/c-count-or-length.html

**Review by ProgrammingIncluded — COMMENTED** [review]
Let's get this in and merge the test fixes from my MR.

File: Assets/Tests/TestBulletPatterns.cs (hunk 1 above)
Comment (ProgrammingIncluded) [reply Assets/Tests/TestBulletPatterns.cs:13]: Fixture for class, sorry wrong location.

File: Assets/Tests/TestBulletPatterns.cs (hunk 2 above)
Comment (NathanielLow) [reply Assets/Tests/TestBulletPatterns.cs:16]: Not on an `IReadOnlyList<T>`. https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.ireadonlylist-1?view=netcore-3.1

File: Assets/Tests/TestBulletPatterns.cs (hunk 1 above)
Comment (NathanielLow) [reply Assets/Tests/TestBulletPatterns.cs:13]: https://docs.nunit.org/articles/nunit/writing-tests/attributes/testfixture.html

Comment (ProgrammingIncluded) [reply Assets/Tests/TestBulletPatterns.cs:13]: Sweet, thanks.

File: Assets/Tests/TestBulletPatterns.cs (hunk 2 above)
Comment (ProgrammingIncluded) [reply Assets/Tests/TestBulletPatterns.cs:16]: Okay, thanks.

**Review by ProgrammingIncluded — APPROVED** [review]

**Review by hirudan — APPROVED** [review]

**Comment by NathanielLow** [comment]
[Add Character Movement Sphere](https://trello.com/c/qePo5LwC/43-add-character-movement-sphere)

**Pull request merged**